### PR TITLE
pd.lib.infer_dtype infers bytes in Python3

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

File: pandas/tests/test_lib.py [hunk 1]
```diff
@@ -2,12 +2,15 @@
...
 
 import numpy as np
+import sys
 
 import pandas as pd
 from pandas.lib import isscalar, item_from_zerodim, max_len_string_array
 import pandas.util.testing as tm
 from pandas.compat import u
 
+PY2 = sys.version_info[0] == 2
```
Comment (jreback) [inline pandas/tests/test_lib.py]: add this to ``pandas.compat`` (or just import ``PY3``)

**Comment by jreback** [comment]
ping!

**Comment by jreback** [comment]
?

**Comment by cpcloud** [comment]
@jreback ok with this?

**Comment by cpcloud** [comment]
oh whoops, need to add tests for strops

**Comment by jreback** [comment]
and add to this line: https://github.com/pydata/pandas/blob/master/pandas/core/base.py#L522

(test if you can)

squash as well. thxs

**Comment by cpcloud** [comment]
ok, in progress

**Comment by cpcloud** [comment]
hm, so to really be solid about this we'd have to test every single str operation against a bytes series

e.g., `cat` was failing because it has a `sep` argument that is `''` if  `None` is passed in. That string will be a unicode string in python 3 which can't be used with a `bytes` object

**Comment by cpcloud** [comment]
Do we have to support `bytes` operations? I really just wanted the ability to tell if a `Series` contained `bytes`. I wasn't looking to add any new features.

**Comment by jreback** [comment]
no I don't think bytes are necessarily a big deal to support - in fact prob should raise on str type ops?

**Comment by cpcloud** [comment]
ok, i'll test that we raise on `str` access

**Comment by jreback** [comment]
ok, looks good. just add a release note and merge. ty.

**Pull request merged**